applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -817,12 +864,108 @@ class TrilinosBlockBuilderAndSolver
         KRATOS_CATCH("");
     }
 
+    /**
+     * @brief Applies the constraints with master-slave relation matrix (RHS only)
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rb The RHS vector
+     */
+    void ApplyRHSConstraints(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        if (rModelPart.GetCommunicator().GlobalNumberOfMasterSlaveConstraints() > 0) {
+            BuildMasterSlaveConstraints(rModelPart);
+
+            // Reference to T
+            const TSystemMatrixType& r_T = *mpT;
+
+            // Compute T' b
+            const TSystemVectorType copy_b(rb);
+            TSparseSpace::TransposeMult(r_T, copy_b, rb);
```

Comment:
this operation may fail, as T^t@b may be of different size than b

also one of the problems is that if we assign and after this rb changed its size, it is not any longer apt for use in assembly

Reply:
T is squared as in serial, we are considering all dofs

Reply:
it could still be larger. Imagine you have a master dof that is not any of the original dofs of the problem

Reply:
That dof is added in the global system as well (see serial implementation)

Reply:
If you have questions about details on implementation, maybe it would be better to speak directly, but I will consider this conversation as resolved.